lightning-invoice/src/payment.rs

```diff
@@ -2161,6 +2067,18 @@ mod tests {
 				panic!("Unexpected amount: {:?}", actual_value_msats);
 			}
 		}
+
+		fn process_route(&self, route: &Route) {
+			for path in &route.paths {
+				self.inflight_htlcs_paths.borrow_mut().push(path.clone());
+			}
+		}
+
+		fn fail_path(&self, path: &Vec<RouteHop>) {
+			if let Some(idx) = self.inflight_htlcs_paths.clone().into_inner().iter().position(|p| p == path) {
```

Comment:
when I tried to do:
```
			if let Some(idx) = self.inflight_htlcs_paths.borrow().iter().position(|p| p == path) {
				self.inflight_htlcs_paths.borrow_mut().swap_remove(idx);
			}
```

I got an error because we're trying to borrow again when doing `swap_remove`, so I took the easy way out here. is there a better way of doing this if we're accessing `inflight_htlcs_paths` as a RefCell both in the `if let` and its body?

Reply:
nvm, figured it out 😅